### PR TITLE
Hw12

### PR DESCRIPTION
**Pull request by E1-Bart0**

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,3 +1,20 @@
+aiohttp==3.7.4.post0
```
Comment (Cjkjvfnby) [inline requirements-dev.txt]: Looks like you copy-pasted the output of the `pip freeze` command.

It is considered a bad practice. Usually, you specify only libs you are using directly like `alembic`.  

I'd recommend having 2 files with requirements for the project: one with requirements to run the project (alembic, etc) and the other for development staff (tests, linter, formatters).  So for development, you need to install from both of them and for production only from one of them.

File: All_home_works/hw12/migrations/script.py.mako [hunk 2]
```diff
@@ -0,0 +1,24 @@
+"""${message}
```
Comment (Cjkjvfnby) [inline All_home_works/hw12/migrations/script.py.mako:1]: Why do we need this file?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: All_home_works/hw12/migrations/script.py.mako (hunk 2 above)
Comment (E1-Bart0) [reply All_home_works/hw12/migrations/script.py.mako:1]: This file is used to generate new migration scripts.

**Comment by E1-Bart0** [comment]
Is it ok, or it needs some changes?

File: All_home_works/hw12/migrations/env.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    This configures the context with just a URL
+    and not an Engine, though an Engine is acceptable
+    here as well.  By skipping the Engine creation
+    we don't even need a DBAPI to be available.
+
+    Calls to context.execute() here emit the given string to the
+    script output.
+
+    """
+    # url = config.get_main_option("sqlalchemy.url")
```
Comment (Cjkjvfnby) [inline All_home_works/hw12/migrations/env.py]: ?

File: All_home_works/hw12/migrations/versions/ca54c2de187c_data_migration.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    user = Student(id=1, first_name="Jon", last_name="Doe")
+    teacher = Teacher(id=1, first_name="Jane", last_name="Doe")
+    homework1 = Homework(id=1, text="First", deadline=2)
+    homework_result = HomeworkResult(
+        id=1, solution="SOLUTION", author=user, homework=homework1, teacher_id=1
+    )
+    session.add_all([user, teacher, homework1, homework_result])
+
+    session.commit()
+    session.close()
```
Comment (Cjkjvfnby) [inline All_home_works/hw12/migrations/versions/ca54c2de187c_data_migration.py]: Does session support `with`?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Look good. Some minor comments.

File: All_home_works/hw12/core/__init__.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from All_home_works.hw12.core.db import URL, Base, Session, engine, session_scope
+
+__all__ = [
```
Comment (Cjkjvfnby) [inline All_home_works/hw12/core/__init__.py]: This is required for asterisks import and asterisks import is not recommended. I prefer not to declare this variable at all.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**